protocol/deposits-interop.md

```diff
@@ -0,0 +1,189 @@
+way to map the event to the deposit transaction without needing to reference any external state.
+
+The injective function is defined by:
+
+$$f:A→B$$
+
+The event emitted from the `OptimismPortal` is defined as:
+
+```solidity
+event TransactionDeposited(address indexed from, address indexed to, uint256 indexed version, bytes opaqueData);
+```
+
+The `DepositTx` is defined as:
+
+```go
+type DepositTx struct {
+	// SourceHash uniquely identifies the source of the deposit
+	SourceHash common.Hash
+	// From is exposed through the types.Signer, not through TxData
+	From common.Address
+	// nil means contract creation
+	To *common.Address `rlp:"nil"`
+	// Mint is minted on L2, locked on L1, nil if no minting.
+	Mint *big.Int `rlp:"nil"`
+	// Value is transferred from L2 balance, executed after Mint (if any)
+	Value *big.Int
+	// gas limit
+	Gas uint64
+	// Field indicating if this transaction is exempt from the L2 gas limit.
+	IsSystemTransaction bool
+	// Normal Tx data
+	Data []byte
+}
+```
+
+In earlier iterations of the interop design, there was an "only EOA invariant" that was used to enable static
+analysis of transactions. By only allowing for EOAs to send cross chain transactions, this meant that the
+calldata for the `CrossL2Inbox` would be present in the transaction's data itself. This could be statically
+analyzed by both the protocol (fork choice rule) and block builders. Forcing EOAs to send the executing
+messages was the biggest complaint by people as it greatly reduced the design space, prevented compatibility
+with smart contract wallets and prevented compatibility with [ERC-4337](https://eips.ethereum.org/EIPS/eip-4337) bundlers.
+
+With the removal of the "only EOA invariant", execution is now a requirement to determine the validity of a transaction.
+This applies for deposit transactions as well. The derivation pipeline needs some sort of legibility into the validity
+of these deposit transactions.
+
+To prevent race conditions on deposited executing messages that reference recent initiating messages, a delay should
+be enforced on deposits that contain executing messages. This prevents attacks where some amount of p2p latency
+can cause nodes to inconsistent chains.
+
+# Proposed Solution
+
+The stateless injective function for mapping deposit events into deposit transactions is updated to be a
+2-valued function, meaning that any input can be mapped to one of two inputs. In mathematical terms:
+
+$$f:A→\{b1,b2\}$$
+
+A deposit transaction event can be mapped to either a valid deposit transaction or a noop deposit transaction.
+
+A new invariant called the "sequencing window invariant" is introduced that enforces a delay for deposit
+transactions that trigger executing messages.
+
+## Derivation Pipeline Execution
+
+The derivation pipeline is updated to include EVM execution of the deposit transactions. A forking EVM provider backed
```

Comment:
We don't want to add execution to the derivation pipeline. We instead want to use a 2 step process that registers the dependency into the L2 state and then force deposits to use state that has been registered ahead of time. This means that there needs to be an "is deposit" introspection from within the EVM. This can be accomplished by having 2 system txs, the first on toggles on `isDeposit() -> true` and the second `isDeposit() -> false`. Ideally we make `isDeposit` only callable by the `CrossL2Inbox` so that arbitrary applications cannot use it to censor